libs/kasync/src/sync/wait_queue.rs

```diff
@@ -934,7 +934,8 @@ impl Waiter {
         // remove the node
         // safety: we have the lock on the queue, so this is safe.
         unsafe {
-            waiters.cursor_from_ptr_mut(ptr).remove();
+            let waiter = Waiter::from_ptr(ptr);
+            waiters.remove(waiter);
```

Comment:
```suggestion
            waiters.remove(ptr);
```

`ptr` is already a `NonNull<Waiter>`